src/systems/a320_systems/src/hydraulic.rs

```diff
@@ -503,137 +556,146 @@ impl A320Hydraulic {
-            .update(self.yellow_loop.pressure());
-        self.engine_driven_pump_2_controller.update(
-            overhead_panel,
-            engine_fire_push_buttons,
-            engine2.uncorrected_n2(),
-            engine2.oil_pressure(),
-            self.engine_driven_pump_2_pressure_switch.is_pressurised(),
-            lgciu2,
-        );
+            self.engine_driven_pump_1_pressure_switch
+                .update(self.green_loop.pressure());
+            self.engine_driven_pump_1_controller.update(
+                overhead_panel,
+                engine_fire_push_buttons,
+                engine1.uncorrected_n2(),
+                engine1.oil_pressure(),
+                self.engine_driven_pump_1_pressure_switch.is_pressurised(),
+                lgciu1,
+            );
 
-        self.engine_driven_pump_2.update(
-            context,
-            &self.yellow_loop,
-            engine2
-                .hydraulic_pump_output_speed()
-                .get::<revolution_per_minute>(),
-            &self.engine_driven_pump_2_controller,
-        );
+            self.engine_driven_pump_1.update(
+                context,
+                &self.green_loop,
+                engine1
+                    .hydraulic_pump_output_speed()
+                    .get::<revolution_per_minute>(),
+                &self.engine_driven_pump_1_controller,
+            );
 
-        self.blue_electric_pump_controller.update(
-            overhead_panel,
-            self.blue_loop.is_pressurised(),
-            engine1.oil_pressure(),
-            engine2.oil_pressure(),
-            engine1.is_above_minimum_idle(),
-            engine2.is_above_minimum_idle(),
-            lgciu1,
-            lgciu2,
-        );
-        self.blue_electric_pump.update(
-            context,
-            &self.blue_loop,
-            &self.blue_electric_pump_controller,
-        );
+            self.engine_driven_pump_2_pressure_switch
+                .update(self.yellow_loop.pressure());
+            self.engine_driven_pump_2_controller.update(
+                overhead_panel,
+                engine_fire_push_buttons,
+                engine2.uncorrected_n2(),
+                engine2.oil_pressure(),
+                self.engine_driven_pump_2_pressure_switch.is_pressurised(),
+                lgciu2,
+            );
 
-        self.yellow_electric_pump_controller.update(
-            context,
-            overhead_panel,
-            &self.forward_cargo_door,
-            &self.aft_cargo_door,
-            self.yellow_loop.is_pressurised(),
-        );
-        self.yellow_electric_pump.update(
-            context,
-            &self.yellow_loop,
-            &self.yellow_electric_pump_controller,
-        );
+            self.engine_driven_pump_2.update(
+                context,
+                &self.yellow_loop,
+                engine2
+                    .hydraulic_pump_output_speed()
+                    .get::<revolution_per_minute>(),
```

Comment:
**suggestion**: The actual measurement unit used is an internal concern of the `EngineDrivePump`. Thus I would suggest passing the `AngularVelocity` itself and only inside the `EngineDrivenPump` and `Pump` converting it to the quantity you need (basically at the point of calculation).

```suggestion
                engine2.hydraulic_pump_output_speed()
```

Reply:
@crocket63 Unresolving, as `Pump` still receives an `f64`.

Reply:
Do you mean in hydraulic/mod.rs ?  I did not go further down as this should all be updated in the next overhaul PR

Reply:
Yes, as a general rule always try to pass `uom` types until the moment you need to get the actual measurement unit in a calculation.